app/src/main/java/de/triology/cas/ldap/CesAuthenticationEventExecutionPlanConfiguration.java

```diff
@@ -0,0 +1,226 @@
+            handler.setPrincipalDnAttributeName(l.getPrincipalDnAttributeName());
+        }
+        handler.setAllowMultiplePrincipalAttributeValues(l.isAllowMultiplePrincipalAttributeValues());
+        handler.setAllowMissingPrincipalAttributeValue(l.isAllowMissingPrincipalAttributeValue());
+        handler.setPasswordEncoder(PasswordEncoderUtils.newPasswordEncoder(l.getPasswordEncoder(), applicationContext));
+        handler.setPrincipalNameTransformer(PrincipalNameTransformerUtils.newPrincipalNameTransformer(l.getPrincipalTransformation()));
+
+        if (StringUtils.isNotBlank(l.getCredentialCriteria())) {
+            LOGGER.trace("Ldap authentication for [{}] is filtering credentials by [{}]",
+                    l.getLdapUrl(), l.getCredentialCriteria());
+            handler.setCredentialSelectionPredicate(CoreAuthenticationUtils.newCredentialSelectionPredicate(l.getCredentialCriteria()));
+        }
+
+        if (StringUtils.isBlank(l.getPrincipalAttributeId())) {
+            LOGGER.trace("No principal id attribute is found for LDAP authentication via [{}]", l.getLdapUrl());
+        } else {
+            handler.setPrincipalIdAttribute(l.getPrincipalAttributeId());
+            LOGGER.trace("Using principal id attribute [{}] for LDAP authentication via [{}]", l.getPrincipalAttributeId(), l.getLdapUrl());
+        }
+
+        val passwordPolicy = l.getPasswordPolicy();
+        if (passwordPolicy.isEnabled()) {
+            LOGGER.trace("Password policy is enabled for [{}]. Constructing password policy configuration", l.getLdapUrl());
+            val cfg = createLdapPasswordPolicyConfiguration(passwordPolicy, authenticator, multiMapAttributes);
+            handler.setPasswordPolicyConfiguration(cfg);
+        }
+
+        val attributes = CollectionUtils.wrap(multiMapAttributes);
+        handler.setPrincipalAttributeMap(attributes);
+
+        LOGGER.debug("Initializing LDAP authentication handler for [{}]", l.getLdapUrl());
+        handler.initialize();
+
+        return handler;
+    }
+
+    @Override
+    public void configureAuthenticationExecutionPlan(AuthenticationEventExecutionPlan plan) {
+        plan.registerAuthenticationHandler(cesGroupAwareLdapAuthenticationHandler());
+        LOGGER.trace("Registered {}, registered authentication handlers: {}",
+                CesGroupAwareLdapAuthenticationHandler.class.getSimpleName(), plan.getAuthenticationHandlers());
+    }
+
+    // 1:1 copy from LdapAuthenticationConfiguration
+    private static PasswordPolicyContext createLdapPasswordPolicyConfiguration(final LdapPasswordPolicyProperties passwordPolicy,
+                                                                               final Authenticator authenticator,
+                                                                               final Multimap<String, Object> attributes) {
+        val cfg = new PasswordPolicyContext(passwordPolicy);
```

Comment:
Continuity, naming:

Following the method name, this method produces a configuration. But following the produced type, this method produces a context. You should provide a stringent naming.

Reply:
In principle, I agree with you. 

However, this code is an exact 1:1 copy of the CAS. I only copied this code because it was not possible to call this method because the method is private. This method should actually be public in the CAS. There should simply be no changes here. 
This is not a great solution, but there is probably no better one at the moment.

That's why I'm leaving this method unchanged.